src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,96 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] values;
+    private int size;
+
+    public ArrayList() {
+        values = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (values.length == size) {
+            T[] sourceArray = (T[]) values;
+            values = new Object[size + (size >> 1)];
+            System.arraycopy(sourceArray, 0, values, 0, size);
```

Comment:
_array should grow 1.5 times when it is full_ - from the task
[Common mistakes](https://mate-academy.github.io/jv-program-common-mistakes/java-core/collections/array-list.html)
**Don’t create repeating code.**
If the logic of your code repeats - move it to the separate private method.
Add `private T[] grow` method